pkg/didcomm/protocol/didexchange/service.go

```diff
@@ -38,29 +49,231 @@ type Service struct {
 	store             storage.Store
 }
 
+//state defines a didcomm service state
+type state struct {
+	MsgType string            `json:"@type"`
+	ID      string            `json:"@id"`
+	Thread  *decorator.Thread `json:"~thread"`
```

Comment:
I think the thread's `thhid` should be enough, not the whole structure.

Reply:
replaced it with thid only string type